ui/src/app/pages/admin/admin-user-profile.spec.tsx

```diff
@@ -491,4 +500,248 @@ describe('AdminUserProfile', () => {
+      AccessModule.ERACOMMONS,
+      {
+        moduleName: AccessModule.ERACOMMONS,
+        bypassEpochMillis: 1,
+      },
+    ],
+    [
+      AccessRenewalStatus.EXPIRING_SOON,
+      AccessModule.PROFILECONFIRMATION,
+      {
+        moduleName: AccessModule.PROFILECONFIRMATION,
+        completionEpochMillis: nowPlusDays(-1000),
+        expirationEpochMillis: nowPlusDays(5),
+      },
+    ],
+  ])(
+    "should render a(n) '%s' completion status for access module %s",
+    async (
+      expectedStatus: AccessRenewalStatus,
+      moduleName: AccessModule,
+      moduleStatus: AccessModuleStatus
+    ) => {
+      const statusesExceptThisOne =
+        TARGET_USER_PROFILE.accessModules.modules.filter(
+          (s) => s.moduleName !== moduleName
+        );
+      updateTargetProfile({
+        accessModules: {
+          ...TARGET_USER_PROFILE.accessModules,
+          modules: [...statusesExceptThisOne, moduleStatus],
+        },
+      });
+      const wrapper = component();
+      expect(wrapper).toBeTruthy();
+      await waitOneTickAndUpdate(wrapper);
+
+      const tableRows = wrapper
+        .find('[data-test-id="access-module-table"]')
+        .find('.p-datatable-row');
+      expect(tableRows.length).toEqual(accessModulesForTable.length);
+
+      // the previous test confirmed that the accessModulesForTable are in the expected order, so we can ref by index
+
+      const { adminPageTitle } = getAccessModuleConfig(moduleName);
+      const moduleRow = tableRows.at(accessModulesForTable.indexOf(moduleName));
+      // sanity check - this is actually the right row for this module
+      expect(
+        findNodesContainingText(moduleRow, adminPageTitle).exists()
+      ).toBeTruthy();
+
+      expect(
+        findNodesContainingText(moduleRow, expectedStatus).exists()
+      ).toBeTruthy();
+    }
+  );
+
+  it('should skip access modules when they are disabled in the environment', async () => {
+    serverConfigStore.set({
+      config: {
+        ...defaultServerConfig,
+        enableRasLoginGovLinking: false,
+        enforceRasLoginGovLinking: false,
+        enableEraCommons: false,
+        enableComplianceTraining: false,
+      },
+    });
+
+    const excludedModules = [
+      AccessModule.RASLINKLOGINGOV,
+      AccessModule.ERACOMMONS,
+      AccessModule.COMPLIANCETRAINING,
+      AccessModule.CTCOMPLIANCETRAINING,
+    ];
+    const expectedModules = accessModulesForTable.filter(
+      (moduleName) => !excludedModules.includes(moduleName)
+    );
+
+    const wrapper = component();
+    expect(wrapper).toBeTruthy();
+    await waitOneTickAndUpdate(wrapper);
+
+    const tableRows = wrapper
+      .find('[data-test-id="access-module-table"]')
+      .find('.p-datatable-row');
+    expect(tableRows.length).toEqual(expectedModules.length);
+
+    // confirm that the expectedModules are listed in order with expected title text
+
+    expectedModules.forEach((moduleName, index) => {
```

Comment:
i wonder if we should extract this logic to a method, since its been used by 2 test already